server/src/main/java/io/crate/replication/logical/analyze/SubscribedTableStatementValidator.java

```diff
@@ -0,0 +1,195 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.replication.logical.analyze;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.AnalyzedAlterTable;
+import io.crate.analyze.AnalyzedAlterTableAddColumn;
+import io.crate.analyze.AnalyzedAlterTableDropCheckConstraint;
+import io.crate.analyze.AnalyzedAlterTableOpenClose;
+import io.crate.analyze.AnalyzedAlterTableRename;
+import io.crate.analyze.AnalyzedCopyFrom;
+import io.crate.analyze.AnalyzedDeleteStatement;
+import io.crate.analyze.AnalyzedDropTable;
+import io.crate.analyze.AnalyzedInsertStatement;
+import io.crate.analyze.AnalyzedStatement;
+import io.crate.analyze.AnalyzedStatementVisitor;
+import io.crate.analyze.AnalyzedSwapTable;
+import io.crate.analyze.AnalyzedUpdateStatement;
+import io.crate.analyze.validator.StatementValidator;
+import io.crate.exceptions.InvalidArgumentException;
+import io.crate.metadata.RelationName;
+import io.crate.user.UserLookup;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.Locale;
+
+import static io.crate.replication.logical.LogicalReplicationSettings.REPLICATION_SUBSCRIPTION_NAME;
+
+public class SubscribedTableStatementValidator implements StatementValidator {
+
+    private static final Validator VALIDATOR = new Validator();
+
+    @Override
+    public void ensureMayExecute(AnalyzedStatement statement,
+                                 UserLookup userLookup,
+                                 SessionContext sessionContext) {
+        statement.accept(VALIDATOR, null);
+    }
+
+    private static class Validator extends AnalyzedStatementVisitor<Void, Void> {
+
+        private static final String DDL_ERROR_MESSAGE = "Table '%s' is replicating by subscription '%s'," +
+                                                        " alter table is not allowed";
+
+        private static final String DROP_ERROR_MESSAGE = "Table table '%s' is replicating by a subscription and " +
+                                                         "cannot be dropped, the subscription '%s' must be dropped first";
+
+        private static final String DML_ERROR_MESSAGE = "Table '%s' is replicating by subscription '%s'," +
+                                                        " writing is not allowed";
+
+        @Override
+        public Void visitAlterTable(AnalyzedAlterTable alterTable, Void context) {
+            // TODO: Validate based on the given settings, some setting must be allowed (e.g. number_of_replicas)
```

Comment:
As there is work ongoing to filter out specific settings while replicating, I did not to the reverse here yet as we could simple re-use the list of filtered settings here.